### PR TITLE
Added Search struct to search URLs for a user

### PR DESCRIPTION
**Pull request by dawntomm**
### Description
Added Search struct to search URLs for a user.

User should be able to search the public and the private short links they created. In order to support that, we add RESTful API to search for all urls created by a given user. The second step is to add Search struct, which provides function to search urls for a user.

### CODE REVIEW

File: backend/app/usecase/search/search.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// Search fetches URLs and users from persistent
+// storage, such as database
+type Search struct {
+	userURLRepo repository.UserURLRelation
+	urlRepo     repository.URL
+}
+
+// SearchForURLs fetches all URLs for a given user
+func (s Search) SearchForURLs(user entity.User) ([]entity.URL, error) {
+	aliases, err := s.userURLRepo.FindAliasesByUser(user)
```
Comment (alldroll) [inline backend/app/usecase/search/search.go]: I do concern about the possible high load problems. I don't know we have restrictions for a number of links created by a user, but how to be if a user has 10k links?

I suggest implementing `paginator` in order to make this potential problem less impact.
What are your thoughts about it?

File: backend/app/usecase/search/search_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			name:         "no such user",
+			user:         user_2,
+			expectedURLs: emptyURLs,
+			expectedErr:  nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			searchAPI := NewSearch(&urlRepo, &userUrlRepo)
```
Comment (alldroll) [inline backend/app/usecase/search/search_test.go]: let's initialize `&urlRepo, &userUrlRepo` inside `Run`.

**Comment by byliuyang** [comment]
#260

**Comment by byliuyang** [comment]
Please support pagination: https://github.com/short-d/short/issues/260#issuecomment-544273964

**Comment by byliuyang** [comment]
Hi @dawntomm, the code coverage is dropping.

**Comment by byliuyang** [comment]
Closing this PR because the APIs will be redesigned to support pagination.

**Pull request closed without merging**